contracts/CommunityTokenDeployer.sol

```diff
@@ -0,0 +1,140 @@
+// SPDX-License-Identifier: UNLICENSED
+
+pragma solidity ^0.8.17;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { ECDSA } from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import { EIP712 } from "@openzeppelin/contracts/utils/cryptography/EIP712.sol";
+import { OwnerToken } from "./OwnerToken.sol";
+import { MasterToken } from "./MasterToken.sol";
+
+contract CommunityTokenDeployer is EIP712("CommunityTokenDeployer", "1"), Ownable {
+    using ECDSA for bytes32;
+
+    event Deploy(address indexed, address indexed);
+
+    error CommunityTokenDeployer_AlreadyDeployed();
+    error CommunityTokenDeployer_InvalidSignerKeyOrCommunityAddress();
+    error CommunityTokenDeployer_InvalidTokenMetadata();
+    error CommunityTokenDeployer_InvalidDeployerAddress();
+    error CommunityTokenDeployer_InvalidDeploymentSignature();
+    error CommunityTokenDeployer_DeploymentSignatureExpired();
+
+    struct TokenConfig {
+        string name;
+        string symbol;
+        string baseURI;
+    }
+
+    struct DeploymentSignature {
+        address signer;
+        address deployer;
+        uint256 deadline;
+        uint8 v;
+        bytes32 r;
+        bytes32 s;
+    }
+
+    bytes32 public constant DEPLOYMENT_SIGNATURE_TYPEHASH =
+        keccak256("Deploy(address signer,address deployer,uint256 deadline)");
+
+    mapping(address => address) public deployments;
+
+    function deploy(
+        TokenConfig calldata _ownerToken,
+        TokenConfig calldata _masterToken,
+        DeploymentSignature calldata _signature,
+        bytes memory _signerPublicKey
+    )
+        external
+        returns (address, address)
+    {
+        if (
+            bytes(_ownerToken.name).length == 0 || bytes(_ownerToken.symbol).length == 0
+                || bytes(_ownerToken.baseURI).length == 0 || bytes(_masterToken.name).length == 0
+                || bytes(_masterToken.symbol).length == 0 || bytes(_masterToken.baseURI).length == 0
+        ) {
+            revert CommunityTokenDeployer_InvalidTokenMetadata();
+        }
```

Comment:
Although `EIP721` token metadata is optional, I think it's still good to require it here as Status clients make heavy use of metadata.